arcgis-ios-sdk-samples/Scenes/Animate images with image overlay/AnimateImagesWithImageOverlayViewController.swift

```diff
@@ -0,0 +1,215 @@
+        // Get the URLs to images added to the project's folder reference.
+        var imageURLs = Bundle.main.urls(forResourcesWithExtension: "png", subdirectory: "PacificSouthWest") ?? []
+        imageURLs.sort { $0.lastPathComponent < $1.lastPathComponent }
+        return CircularIterator(
+            elements: imageURLs.map { UIImage(contentsOfFile: $0.path)! }
+        )
+    }()
+    
+    /// A formatter to format percentage strings.
+    let percentageFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .percent
+        formatter.multiplier = 100
+        return formatter
+    }()
+    
+    /// An envelope of the pacific southwest sector for displaying the image frame.
+    let pacificSouthwestEnvelope = AGSEnvelope(
+        center: AGSPoint(
+            x: -120.0724273439448,
+            y: 35.131016955536694,
+            spatialReference: .wgs84()
+        ),
+        width: 15.09589635986124,
+        height: -14.3770441522488
+    )
+    
+    // MARK: Initialize scene and set image frame
+    
+    /// Create a scene.
+    ///
+    /// - Returns: A new `AGSScene` object.
+    func makeScene() -> AGSScene {
+        // Create a tiled layer from World Dark Gray Base REST service.
+        let basemapTileURL = URL(string: "https://services.arcgisonline.com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer")!
+        let worldDarkGrayBasemap = AGSArcGISTiledLayer(url: basemapTileURL)
+        // Create an elevation source from Terrain3D REST service.
+        let elevationServiceURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+        let elevationSource = AGSArcGISTiledElevationSource(url: elevationServiceURL)
+        // Create a surface and add it to the scene.
+        let surface = AGSSurface()
+        surface.elevationSources = [elevationSource]
+        let scene = AGSScene(basemap: AGSBasemap(baseLayer: worldDarkGrayBasemap))
+        scene.baseSurface = surface
+        return scene
+    }
+    
+    /// Set current image to the image overlay.
+    @objc
+    func setImageFrame() {
+        let frame = AGSImageFrame(image: imagesIterator.next()!, extent: pacificSouthwestEnvelope)
+        imageOverlay.imageFrame = frame
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func sliderValueChanged(_ slider: UISlider) {
+        imageOverlay.opacity = slider.value
+        opacityLabel.text = percentageFormatter.string(from: slider.value as NSNumber)!
+    }
+    
+    @IBAction func playPauseButtonTapped(_ button: UIBarButtonItem) {
+        let index = toolbar.items!.firstIndex(of: button)!
+        if !displaylink.isPaused {
+            toolbar.items![index] = playButtonItem
+        } else {
+            toolbar.items![index] = pauseButtonItem
+            // Set framerate of the display link before starting to play.
+            displaylink.preferredFramesPerSecond = frameRate
```

Comment:
Does the display link require that this property be set before being paused? If not, setting it each time the animation is resumed it overkill. It really should only be set once when the user changes the speed.